index.js

```diff
@@ -38,73 +38,116 @@ function f(d) {
+        loading = false;
+      }
+    },
     onRendered: (result) => {
       const d = result.minDistance;
       const handBox = result.handBox;
       const faceBox = result.faceBox;
       const deltaVolume = result.deltaVolume;
 
-      document.querySelector('#distance').innerText = d
-        ? `Closest ||p||: ${f(d.d)}, Δx: ${f(d.x)}, Δy: ${f(d.y)}, Δz: ${f(d.z)}`
-        : `Closest ||p||: Undefined`;
+      // document.querySelector('#distance').innerText = d
+      //   ? `Closest ||p||: ${f(d.d)}, Δx: ${f(d.x)}, Δy: ${f(d.y)}, Δz: ${f(d.z)}`
+      //   : `Closest ||p||: Undefined`;
 
-      document.querySelector('#intersection').innerText =
-        `Volume intersected: ${deltaVolume}`;
-      document.querySelector('#deltaCenter').innerText = d
-        ? `Center bounding box ||p||: ${f(Math.sqrt(Math.pow(handBox.xCenter - faceBox.xCenter, 2) + Math.pow(handBox.yCenter - faceBox.yCenter, 2) + Math.pow(handBox.zCenter - faceBox.zCenter, 2)))} Δx:${f(handBox.xCenter - faceBox.xCenter)} Δy:${f(handBox.yCenter - faceBox.yCenter)} Δz:${f(handBox.zCenter - faceBox.zCenter)}`
-        : `Center bounding box: Undefined`;
+      // document.querySelector('#intersection').innerText =
+      //   `Volume intersected: ${deltaVolume}`;
+      // document.querySelector('#deltaCenter').innerText = d
+      //   ? `Center bounding box ||p||: ${f(Math.sqrt(Math.pow(handBox.xCenter - faceBox.xCenter, 2) + Math.pow(handBox.yCenter - faceBox.yCenter, 2) + Math.pow(handBox.zCenter - faceBox.zCenter, 2)))} Δx:${f(handBox.xCenter - faceBox.xCenter)} Δy:${f(handBox.yCenter - faceBox.yCenter)} Δz:${f(handBox.zCenter - faceBox.zCenter)}`
+      //   : `Center bounding box: Undefined`;
 
-      document.querySelector('#detection').innerText =
-        `Detection: ${result.detected ? 'Yes' : 'No'}`;
+      // document.querySelector('#detection').innerText =
+      //   `Detection: ${result.detected ? 'Yes' : 'No'}`;
+      faceCurrentlyTouched = result.detected;
       stats.end();
     }
   };
 
-  // Check every 5 secs with at least three touches
+  // Check every second with at least three touches
   workerTimers.setInterval(() => {
-    if (touchCounter > 2 && Notification.permission === 'granted') {
-      new Notification('You touched your face!');
+    if (touchCounter >= 2 && Notification.permission === 'granted' && !faceAlreadyTouched) {
+      new Notification('🤭 You touched your face! 🤭');
+      $("#face-touch-alert").show();
+      totalTouches++;
+      document.querySelector('#totalCount').innerText = totalTouches;
+      document.querySelector('#timesTouchedText').innerText = totalTouches === 1 ? 'time touched' : 'times touched';
+      window.document.title = '😱'
+      // TODO: add log! stretch: take canvas snapshots for each...
+      let today = new Date();
+      let time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
+      console.log(time)
```

Comment:
Remove unused code. In my heat map work, I create a history class to store history, I think we can move all the history related stuff to that class